src/bucket/Bucket.cpp

```diff
@@ -888,13 +890,21 @@ Bucket::scanForEviction(AbstractLedgerTxn& ltx, EvictionIterator& iter,
                     }
 
                     releaseAssert(ttlLtxe);
-
+                    liveUntilLedger =
+                        ttlLtxe.current().data.ttl().liveUntilLedgerSeq;
                     return !isLive(ttlLtxe.current(), ledgerSeq);
                 };
 
                 if (shouldEvict())
                 {
                     ZoneNamedN(evict, "evict entry", true);
+                    if (metrics.has_value())
+                    {
+                        ++metrics->numEntriesEvicted;
+                        metrics->evictedEntriesAgeSum +=
```

Comment:
why do we need to sum it up like this? we could use a histrogram (that would live inside `Bucket`), and update it every time we evict a single entry instead of manually aggregating evictions.

Reply:
We actually don't want a histogram and specifically want the per-cycle aggregation. A histogram for this metric is misleading. Given the nature of the eviction scan, entries evicted in level 8 will almost certainly be "older" than entries evicted in level 6. Within a given cycle, the number of evicted entries is very noisy from level to level. We want to abstract away that noise. We only care if the average age increases cycle-to-cycle. This would be pretty involved to graph in Grafana if we use a histogram, as the cycle length itself is variable. This seemed significantly easier, as it logs the only metric we actually care about (average cycle-to-cycle) in a way that is easy to analyze in Grafana.